Fix build of Box2D with gcc4.3.
Still applies against current Box2D upstream.

Lubomir Rintel <lkundrak@v3.sk>

diff -urp numptyphysics/Box2D/Source/Collision/b2BroadPhase.cpp numptyphysics/Box2D/Source/Collision/b2BroadPhase.cpp
--- numptyphysics.orig/Box2D/Source/Collision/b2BroadPhase.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Box2D/Source/Collision/b2BroadPhase.cpp	2008-09-25 16:16:39.000000000 +0200
@@ -18,6 +18,7 @@
 
 #include "b2BroadPhase.h"
 #include <algorithm>
+#include <cstring>
 
 // Notes:
 // - we use bound arrays instead of linked lists for cache coherence.
diff -urp numptyphysics/Box2D/Source/Common/b2BlockAllocator.cpp numptyphysics/Box2D/Source/Common/b2BlockAllocator.cpp
--- numptyphysics.orig/Box2D/Source/Common/b2BlockAllocator.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Box2D/Source/Common/b2BlockAllocator.cpp	2008-09-25 16:17:46.000000000 +0200
@@ -20,6 +20,7 @@
 #include <cstdlib>
 #include <memory>
 #include <climits>
+#include <cstring>
 
 int32 b2BlockAllocator::s_blockSizes[b2_blockSizes] = 
 {
diff -urp numptyphysics/Box2D/Source/Dynamics/Contacts/b2CircleContact.cpp numptyphysics/Box2D/Source/Dynamics/Contacts/b2CircleContact.cpp
--- numptyphysics.orig/Box2D/Source/Dynamics/Contacts/b2CircleContact.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Box2D/Source/Dynamics/Contacts/b2CircleContact.cpp	2008-09-25 16:17:57.000000000 +0200
@@ -22,6 +22,7 @@
 #include "../../Common/b2BlockAllocator.h"
 
 #include <new>
+#include <cstring>
 
 b2Contact* b2CircleContact::Create(b2Shape* shape1, b2Shape* shape2, b2BlockAllocator* allocator)
 {
diff -urp numptyphysics/Box2D/Source/Dynamics/Contacts/b2PolyAndCircleContact.cpp numptyphysics/Box2D/Source/Dynamics/Contacts/b2PolyAndCircleContact.cpp
--- numptyphysics.orig/Box2D/Source/Dynamics/Contacts/b2PolyAndCircleContact.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Box2D/Source/Dynamics/Contacts/b2PolyAndCircleContact.cpp	2008-09-25 16:17:48.000000000 +0200
@@ -22,6 +22,7 @@
 #include "../../Common/b2BlockAllocator.h"
 
 #include <new>
+#include <cstring>
 
 b2Contact* b2PolyAndCircleContact::Create(b2Shape* shape1, b2Shape* shape2, b2BlockAllocator* allocator)
 {
diff -urp numptyphysics/Box2D/Source/Dynamics/Contacts/b2PolyContact.cpp numptyphysics/Box2D/Source/Dynamics/Contacts/b2PolyContact.cpp
--- numptyphysics.orig/Box2D/Source/Dynamics/Contacts/b2PolyContact.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Box2D/Source/Dynamics/Contacts/b2PolyContact.cpp	2008-09-25 16:17:49.000000000 +0200
@@ -23,6 +23,7 @@
 
 #include <memory>
 #include <new>
+#include <cstring>
 
 b2Contact* b2PolygonContact::Create(b2Shape* shape1, b2Shape* shape2, b2BlockAllocator* allocator)
 {
diff -urp numptyphysics/Http.cpp numptyphysics/Http.cpp
--- numptyphysics.orig/Http.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Http.cpp	2008-09-25 16:13:39.000000000 +0200
@@ -14,6 +14,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 
diff -urp numptyphysics/Levels.cpp numptyphysics/Levels.cpp
--- numptyphysics.orig/Levels.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Levels.cpp	2008-09-25 16:13:39.000000000 +0200
@@ -15,6 +15,7 @@
  */
 
 
+#include <string.h>
 #include <sys/types.h>
 #include <dirent.h>
 
diff -urp numptyphysics/Path.cpp numptyphysics/Path.cpp
--- numptyphysics.orig/Path.cpp	2008-09-25 14:40:28.000000000 +0200
+++ numptyphysics/Path.cpp	2008-09-25 16:13:39.000000000 +0200
@@ -15,6 +15,7 @@
  */
 
 
+#include <string.h>
 #include "Path.h"
 
 
